Game/Scripts/SwitchPlayerManagerScript.cpp

```diff
@@ -0,0 +1,226 @@
+#include "StdAfx.h"
+#include "SwitchPlayerManagerScript.h"
+
+#include "ModulePlayer.h"
+#include "ModuleInput.h"
+#include "ModuleScene.h"
+
+#include "Components/ComponentScript.h"
+#include "Components/ComponentTransform.h"
+#include "Components/UI/ComponentTransform2D.h"
+#include "Components/ComponentRigidBody.h"
+#include "Components/ComponentPlayer.h"
+#include "Components/ComponentParticleSystem.h"
+#include "Components/ComponentAnimation.h"
+
+#include "../Scripts/PlayerManagerScript.h"
+#include "PlayerAttackScript.h"
+#include "ParticleBillboardAsistance.h"
+
+#include "../Scripts/CameraControllerScript.h"
+
+#include "Application.h"
+#include "Scene/Scene.h"
+
+REGISTERCLASS(SwitchPlayerManagerScript);
+
+SwitchPlayerManagerScript::SwitchPlayerManagerScript() : Script(), camera(nullptr), input(nullptr), isSwitchAvailable(true), changingPlayerTime{ 800.f, 1800.f},
+	playerHealthBar(nullptr), secondPlayerHealthBar(nullptr)
+{
+	REGISTER_FIELD(isSwitchAvailable, bool);
+	REGISTER_FIELD(playerHealthBar, GameObject*);
+	REGISTER_FIELD(secondPlayerHealthBar, GameObject*);
+	REGISTER_FIELD(secondPlayer, GameObject*);
+	REGISTER_FIELD(switchPlayersParticlesPrefab, GameObject*);
+}
+
+void SwitchPlayerManagerScript::Start()
+{
+	input = App->GetModule<ModuleInput>();
+	
+	mainCamera = App->GetModule<ModulePlayer>()->GetCameraPlayerObject();
+
+	camera = mainCamera->GetComponent<CameraControllerScript>();
+	cameraTransform = mainCamera->GetComponent<ComponentTransform>();
+
+	currentPlayer = App->GetModule<ModulePlayer>()->GetPlayer();
+	playerManager = currentPlayer->GetComponent<PlayerManagerScript>();
+
+	camera->ChangeCurrentPlayer(currentPlayer->GetComponent<ComponentTransform>());
+
+	if (switchPlayersParticlesPrefab)
+	{
+		switchPlayersParticlesPrefab->Disable();
+	}
+}
+
+void SwitchPlayerManagerScript::Update(float deltaTime)
+{
+	if (!isChangingPlayer)
+	{
+		if (input->GetKey(SDL_SCANCODE_C) != KeyState::IDLE && secondPlayer 
+			&& currentPlayer->GetComponent<PlayerManagerScript>()->IsGrounded()
+			&& isSwitchAvailable)
+		{
+			CheckChangeCurrentPlayer();
+		}
+	}
+	else 
+	{
+		HandleChangeCurrentPlayer();
+
+		if(isSwitchingHealthBars)
+		{
+			float t = (changePlayerTimer.Read() - changingPlayerTime[0]) / 1000;
```

Comment:
```suggestion
			float t = (changePlayerTimer.Read() - changingPlayerTime[0]) / 1000;
```
probably `t` is not the best name you could give to that variable